pageserver/src/tenant/timeline.rs

```diff
@@ -3045,10 +3102,17 @@ impl Timeline {
                     if is_dup_layer
                         || dup_end_lsn.is_valid()
                         || written_size + key_values_total_size > target_file_size
+                        || (next_hole < holes.len() && key >= holes[next_hole].key_range.end)
+                    // stop on hole
```

Comment:
We can have this as a configuration based decision (if to apply this optimization or not) so we can ship this and test in real tenants

Reply:
> Still valid.

Thank you replaced with unstable sort.
Also add comment for `other_len.cmp(&self_len)` (I prefer it from calling `reverse` function)

Reply:
> We can have this as a configuration based decision (if to apply this optimization or not) so we can ship this and test in real tenants

Can be done, but please notice that it is not short-turnaround configuration change: to check effect of such change we need to recreate database from scratch and better even reproduce original workload.

Reply:
I think this is the only remaining issue. I think we can start testing in staging, but this is something to consider with the next release that the staging should had been monitored. I'll create an issue for the tenantconf difficulties.

Reply:
@shanyp  Do you have objections to merge this PR without feature flag (can be added later?)
If not, please mark this conversation as resolved.

Reply:
Lets test this in staging , I can later on add the feature flag if needed